src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/AbstractLocation.cs

```diff
@@ -134,10 +134,10 @@ public SyntaxNode TryGetNodeToReportDiagnostic(PointsToAnalysisResult pointsToAn
             }
 
             // Fallback to reporting the diagnostic on the allocation location.
-            return CreationOpt.Syntax;
+            return CreationOpt?.Syntax;
 
             // Local functions.
-            SyntaxNode TryGetSyntaxNodeToReportDiagnostic(IOperation creation)
+            SyntaxNode? TryGetSyntaxNodeToReportDiagnostic(IOperation creation)
```

Comment:
:bulb: You can update just this outer local function to take the non-null value, and update the inner local functions to all reference this parameter.

```suggestion
            SyntaxNode? TryGetSyntaxNodeToReportDiagnostic(PointsToAnalysisResult pointsToAnalysisResult, IOperation creation)
```